tests/kernel/mem_protect/syscalls/src/main.c

```diff
@@ -377,6 +437,95 @@ void test_syscall_context(void)
 
 K_MEM_POOL_DEFINE(test_pool, BUF_SIZE, BUF_SIZE, 4 * NR_THREADS, 4);
 
+/**
+ * @brief Test CPU scrubs registers after system call
+ *
+ * @details - Call string nlen and right before exit from
+ * the system call function write into registers 0xDEADBEEF value
+ * - Then in main test function below check registers values,
+ * if no 0xDEADBEEF value detected, that means CPU scrubbed registers
+ * before exit from the system call.
+ *
+ * @ingroup kernel_memprotect_tests
+ */
+void test_syscall_cpu_scrubs_regs(void)
+{
+#if CONFIG_X86
+#if CONFIG_X86_64
+	int err;
+	size_t ret;
+	long x86_64_reg_val[7];
+
+	ret = string_nlen_reg(user_string, BUF_SIZE, &err);
+	__asm__ volatile(
+		"\t movq %%rbx,%0" : "=r"(x86_64_reg_val[0]));
+	__asm__ volatile(
+		"\t movq %%rcx,%0" : "=r"(x86_64_reg_val[1]));
+	__asm__ volatile(
+		"\t movq %%rdx,%0" : "=r"(x86_64_reg_val[2]));
+	__asm__ volatile(
+		"\t movq %%r8,%0" : "=r"(x86_64_reg_val[3]));
+	__asm__ volatile(
+		"\t movq %%r9,%0" : "=r"(x86_64_reg_val[4]));
+	__asm__ volatile(
+		"\t movq %%r10,%0" : "=r"(x86_64_reg_val[5]));
+	__asm__ volatile(
+		"\t movq %%r11,%0" : "=r"(x86_64_reg_val[6]));
+
+	for (int i = 0; i < 7; i++) {
+		zassert_true(x86_64_reg_val[i] != DB_VAL,
+				"register value is 0xDEADBEEF, "
+				"not scrubbed after system call.");
+	}
+#endif
+#endif
+#if CONFIG_X86
+#ifndef CONFIG_X86_64
```

Comment:
How about just #else

Reply:
I'm afraid if to use there #else, branches below will be broken.